aea/registries/base.py

```diff
@@ -225,292 +237,165 @@ def _add_protocol(self, protocol_directory: Path):
         )
 
         # instantiate the protocol manager.
-        protocol = Protocol(protocol_name, serializer, protocol_config)
-        self.register((protocol_name, None), protocol)
+        protocol = Protocol(protocol_config.public_id, serializer, protocol_config)
+        protocol_public_id = PublicId(
+            protocol_config.author, protocol_config.name, protocol_config.version
+        )
+        self.register(protocol_public_id, protocol)
 
 
-class HandlerRegistry(Registry):
-    """This class implements the handlers registry."""
+class ComponentRegistry(
+    Registry[Tuple[SkillId, str], SkillComponentType], Generic[SkillComponentType]
+):
+    """This class implements a generic registry for skill components."""
 
     def __init__(self) -> None:
         """
         Instantiate the registry.
 
         :return: None
         """
-        self._handlers = {}  # type: Dict[ProtocolId, Dict[SkillId, Handler]]
+        self._items = {}  # type: Dict[SkillId, Dict[str, SkillComponentType]]
```

Comment:
the cool thing of a well-defined interface is that we can change the implementation without breaking user (and our) code.

Reply:
e.g. we can later create an index of a different kind.

Reply:
yep!